WMIAdventure/frontend/src/js/components/global/atoms/CompactCardView/CompactCardView.js

```diff
@@ -17,12 +18,16 @@ class CompactCardView extends React.Component {
         setTranslateX -> handle movement animation
         cardLevel -> information about card level
         cardImage -> card icon
+        cardsOrder -> to show particular card in BattleView and hide the rest
+        setOpacity -> to handle opacity animation in BattleView
      */
 
     state = {
         cardName: ' ',
         cardLevel: 1,
-        cardImage: null
+        cardImage: null,
+        compactCardOpacity: '1',
+        cardIndexInDeck: this.props.cardIndexInDeck,

```

Comment:
Czemu to jest tutaj?
Wg mnie to powinien być osobny komponent, bo w edytorze np nie potrzebujemy tych stanów.
Zrobiłbym nowy komponent który będzie dziedziczył z tego

Reply:
Fajnie, że jest reużywalny właśnie, po co robić nowy komponent, skoro można dopisać kilka linijek do istniejącego i jest ok?
Czym to szkodzi, że tak jest?

Reply:
Mamy przeładowany komponent, który już nie jest tylko kompaktową kratą. Nie jest jasne co tak naprawdę ten komponent ma robić. Skoro coś nie jest potrzebne w jednym miejscu to lepiej żeby tego nie było. 
Jakby był osobny komponent to byłoby to dużo bardziej czytelne. Jak ktoś będzie potrzebował użyć tego komponentu, to użyje tego, a jak nie będzie chciał tej funkcjonalności, to użyje bazowego. To też jest reużywalne.

Reply:
Jak zrobisz dziedziczenie z tego komponentu to nie będziesz pisał więcej kodu, bo już cały kod będzie napisane w komponencie bazowym i tutaj tylko dodasz funkcje których potrzebujesz

Reply:
Render praktycznie nie różni się niczym, mogę wywołać `super.render()`, ale jak przekazać do głównego Div te dwa dodatkowe propsy, które potrzebuje, nie pisząc takiego samego kolejnego render od początku?

Reply:
To faktycznie komplikuje trochę sprawę, możnaby rozbić ten render na kilka mniejszych funkcji, no ale nie warto teraz tego robić

Reply:
React słaby framework po prostu XD

Reply:
Dwa opcjonalne argumenty po prostu, które jak dodasz, sprawiają że obiekt umie trochę więcej